packages/jaeger-ui/src/utils/date.tsx

```diff
@@ -40,67 +43,100 @@ const UNIT_STEPS: { unit: string; microseconds: number; ofPrevious: number }[] =
   { unit: 'μs', microseconds: 1, ofPrevious: 1000 },
 ];
 
-const timeUnitToShortTermMapper = {
+type ShortTimeUnit = 'μs' | 'ms' | 's' | 'm' | 'h' | 'd';
+type LongTimeUnit = 'microseconds' | 'milliseconds' | 'seconds' | 'minutes' | 'hours' | 'days';
+
+const timeUnitToShortTermMapper: {
+  [key in LongTimeUnit]: ShortTimeUnit;
+} = {
+  microseconds: 'μs',
   milliseconds: 'ms',
   seconds: 's',
   minutes: 'm',
   hours: 'h',
   days: 'd',
-};
+} as const;
 
 /**
  * @param {number} timestamp
  * @param {number} initialTimestamp
  * @param {number} totalDuration
  * @return {number} 0-100 percentage
  */
-export function getPercentageOfDuration(duration: number, totalDuration: number) {
+export function getPercentageOfDuration(duration: number, totalDuration: number): number {
   return (duration / totalDuration) * 100;
 }
 
-const quantizeDuration = (duration: number, floatPrecision: number, conversionFactor: number) =>
+const quantizeDuration = (duration: number, floatPrecision: number, conversionFactor: number): number =>
   toFloatPrecision(duration / conversionFactor, floatPrecision) * conversionFactor;
 
 /**
- * @param {number} duration (in microseconds)
+ * @param {number} duration - number of microseconds elapsed since midnight, January 1, 1970 Universal Coordinated Time (UTC).
```

Comment:
Some nits - 
You can also replace the comment with unix-time, as it is a well established standard for time.